### PR TITLE
write tests for controllers

### PR DESCRIPTION
**Pull request by huzaifakhan0308**
- Write tests for controllers.

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    end
+  end
+
+  describe 'Response body includes correct placeholder text' do
+    it 'displays the correct placeholder text' do
+      get user_post_path(1, 1)
+      expect(response.body).to include('Here is a list of post for a given user')
+    end
+  end
+end
```
Comment (rotimiazeez) [inline spec/requests/posts_spec.rb:45]: Please some of your tests are not passing, kindly fix this 👍🏽 
Screenshot from 2023-05-13 01-01-36

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi @huzaifakhan0308 👍🏽,
## Good job so far!
### To Highlight 👇🏽
- Has and uses RSpec gem for testing 👍🏽 
- No Linter error ✔️ 
- Good PR description 👍🏽 
- Good commit message(s) 😉
- Descriptive README 📖
You still need to work on some issues to prepare your project for the final evaluation, but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR WORDS>


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag (@rotimiazeez) me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

File: spec/requests/posts_spec.rb (hunk 1 above)
Comment (huzaifakhan0308) [reply spec/requests/posts_spec.rb:45]: as you can see here, in my local machine, all tests are passing.

Screenshot 2023-05-13 124523

**Review by KDlamini — APPROVED** [review]
### Hi @huzaifakhan0308 🙋‍♀️,  

Great work! 💯 
You have done a great job implementing all the project requirements. 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!🎊 🥳 

## Highlights 🌞 

✔️  Good PR title and description
✔️  `index` and `show` actions for `User` and `Post` controllers are tested
✔️  There's `users_spec.rb` and `posts_spec.rb` in `spec/requests/` directory.
✔️  App uses RSpec gem for testing

# Status: Approved ✔️ 

LeonardoInGatsby

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**